### PR TITLE
Feature/sponsor view with api

### PR DESCRIPTION
**Pull request by dominikas**
Added back buttons for all views
Added API Category Views

### CODE REVIEW

File: mobile/fmmApp/fmmApp/DataService/CategoryListDataService.cs [hunk 1]
```diff
@@ -42,22 +40,48 @@ public class CategoryListDataService
         /// <typeparam name="T">Type of view model.</typeparam>
         /// <param name="fileName">Json file to fetch data.</param>
         /// <returns>Returns the view model object.</returns>
-        private static T PopulateData<T>(string fileName)
+
+        public async Task<CategoryListViewModel> PopulateData()
         {
-            var file = "fmmApp.Data." + fileName;
+            var service = Startup.ServiceProvider.GetService<IDataStore<Category>>();
```
Comment (piotrnieweglowski) [inline mobile/fmmApp/fmmApp/DataService/CategoryListDataService.cs:46]: I've noticed that you use var service = Startup.ServiceProvider.GetService<IDataStore<Category>>(); in many places in the class. Maybe let's save a reference to it as a private field.

File: mobile/fmmApp/fmmApp/Controls/CategoryList.cs [hunk 2]
```diff
@@ -20,14 +20,14 @@ public override bool FilterContacts(object obj)
         {
```
Comment (piotrnieweglowski) [inline mobile/fmmApp/fmmApp/Controls/CategoryList.cs:20]: Do we have a chance to rename FilterContacts to let's say just 'Filter'?

File: mobile/fmmApp/fmmApp/DataService/SponsorsListDataService.cs [hunk 3]
```diff
@@ -53,6 +54,30 @@ public async Task<SponsorsListViewModel> PopulateData()
             return viewModel;
         }
 
+        public async Task<bool> SaveNewSponsor(Sponsor sponsor)
+        {
+            var service = Startup.ServiceProvider.GetService<IDataStore<Sponsor>>();
```
Comment (piotrnieweglowski) [inline mobile/fmmApp/fmmApp/DataService/SponsorsListDataService.cs:59]: As in previous comment, I think that it is worth to save a reference to service.

File: mobile/fmmApp/fmmApp/Models/Category.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Runtime.Serialization;
+using Xamarin.Forms.Internals;
+
+namespace fmmApp.Models
+{
+    [ResourcePath("category")]
+    
+    public class Category : ResourceItem<Category>
+    {
+        [DataMember(Name = "description")]
```
Comment (piotrnieweglowski) [inline mobile/fmmApp/fmmApp/Models/Category.cs:11]: probably you can skip DataMember attribute. In FmmClient we have a setting that everything what is PascalCase is transformed and the result is in cammelCase.

File: mobile/fmmApp/fmmApp/Services/FmmClient.cs [hunk 5]
```diff
@@ -126,6 +126,9 @@ private static void SerializeJsonIntoStream(object value, Stream stream)
                 var serializer = new JsonSerializer();
                 serializer.ContractResolver = new CamelCasePropertyNamesContractResolver();
                 serializer.Serialize(jsonTextWriter, value);
+
```
Comment (piotrnieweglowski) [inline mobile/fmmApp/fmmApp/Services/FmmClient.cs:129]: let's remove that, it is not needed in production code.

File: mobile/fmmApp/fmmApp/Views/Forms/AddCategoryPage.xaml.cs [hunk 6]
```diff
@@ -23,10 +25,18 @@ private void CancelButton_Clicked(object sender, EventArgs e)
...
-        private void AddButton_Clicked(object sender, EventArgs e)
+        
+        public async void AddButton_Clicked(object sender, EventArgs e)
         {
-
+            var category = new Category
+            {
+                Description = this.NameEntry.Text
+            };
+            //sprawdzenie, czy na pewno chcesz dodać
```
Comment (piotrnieweglowski) [inline mobile/fmmApp/fmmApp/Views/Forms/AddCategoryPage.xaml.cs:35]: Probably you can remove the comment :)

Comment (dominikas) [reply mobile/fmmApp/fmmApp/Views/Forms/AddCategoryPage.xaml.cs:35]: Done :) However, I think we should add the "feature" which is mentioned in the comment :)

File: mobile/fmmApp/fmmApp/Services/FmmClient.cs (hunk 5 above)
Comment (dominikas) [reply mobile/fmmApp/fmmApp/Services/FmmClient.cs:129]: Done

File: mobile/fmmApp/fmmApp/Models/Category.cs (hunk 4 above)
Comment (dominikas) [reply mobile/fmmApp/fmmApp/Models/Category.cs:11]: Done

File: mobile/fmmApp/fmmApp/DataService/CategoryListDataService.cs (hunk 1 above)
Comment (dominikas) [reply mobile/fmmApp/fmmApp/DataService/CategoryListDataService.cs:46]: I've added a reference (_servce) for all methods except populateData. When I've used _service there, after changing views and getting back to the category list, the data was doubled/trippled etc. Let me know if it is ok for you.

**Pull request merged**